pymc_marketing/model_builder.py

```diff
@@ -341,6 +341,16 @@ def default(x):
             default=default,
         )
 
+        attrs["dag"] = (
+            json.dumps(self.dag, default=default) if hasattr(self, "dag") else "None"
+        )
+        attrs["treatment_nodes"] = (
+            self.treatment_nodes if hasattr(self, "treatment_nodes") else "None"
+        )
+        attrs["outcome_node"] = (
+            self.outcome_node if hasattr(self, "outcome_node") else "None"
+        )
```

Comment:
I do not think this **has** to be in the model builder class, which is much more general. For instance, this is not necessary for the CLV module.

Reply:
100%

Reply:
Then where should I added? I got issues if I don't add them here, but let me try again.

Reply:
ps: This is connect to the test, without this code the test fail.